core/src/test/java/org/elasticsearch/script/ScriptModesTests.java

```diff
@@ -94,6 +97,21 @@ public void assertAllSettingsWereChecked() {
         }
     }
 
+    private static Setting<?>[] buildDeprecatedSettingsArray(ScriptSettings scriptSettings, String... keys) {
```

Comment:
This seems to be common, maybe just put it under one test class as public static?